graylog2-web-interface/src/components/graylog/SplitButton.jsx

```diff
@@ -1,21 +1,23 @@
 import React, { forwardRef, useCallback } from 'react';
+// eslint-disable-next-line no-restricted-imports
 import { SplitButton as BootstrapSplitButton } from 'react-bootstrap';
 import styled from 'styled-components';
 
 import buttonStyles from './styles/button';
 import { propTypes, defaultProps } from './props/button';
 
-const SplitButton = forwardRef(({ active, bsStyle, ...props }, ref) => {
+const SplitButton = forwardRef((props, ref) => {
+  const { active, bsStyle, disabled } = props;
   const StyledSplitButton = useCallback(styled(BootstrapSplitButton)`
-    ${buttonStyles({ active })};
+    ${buttonStyles(props)};
 
     ~ .btn.dropdown-toggle {
-      ${buttonStyles({ active })};
+      ${buttonStyles(props)};
     }
-  `, [active]);
+  `, [active, bsStyle, disabled]);
```

Comment:
If we pass in `props` to the `buttonStyles` function used in the callback, imo we also need to use the whole `props` object as a dependency, because otherwise if the `buttonStyles` changes and we suddenly also use a e.g. `bsSize` prop from it, the callback might get stale if only `bsSize` changes.